client/post-editor/editor-ground-control/history-button.jsx

```diff
@@ -2,36 +2,48 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { localize } from 'i18n-calypso';
-import { flow } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import { openPostRevisionsDialog } from 'state/posts/revisions/actions';
+import { withAnalytics, recordTracksEvent } from 'state/analytics/actions';
+import EditorRevisionsDialog from 'post-editor/editor-revisions/dialog';
 
-const HistoryButton = ( { openDialog, translate } ) => (
-	<div className="editor-ground-control__history">
-		<button className="editor-ground-control__history-button button is-link" onClick={ openDialog }>
-			{ translate( 'History' ) }
-		</button>
-	</div>
-);
+class HistoryButton extends Component {
+	render() {
+		const { loadRevision, postId, siteId, translate } = this.props;
+		return (
+			<div className="editor-ground-control__history">
+				<button
+					className="editor-ground-control__history-button button is-link"
+					onClick={ this.props.selectHistory }
+				>
+					{ translate( 'History' ) }
+				</button>
+				<EditorRevisionsDialog loadRevision={ loadRevision } postId={ postId } siteId={ siteId } />
+			</div>
+		);
+	}
+}
```

Comment:
also we can note that at this point we've come full-circle and realize we could easily stick with the function component…

Reply:
Btw, is there any real benefit making it `functional` vs `React.PureComponent`?

Reply:
it depends on your vantage point. in some insignificant ways it's less code we're shipping, though those can add up across the app. it's nice for testing and easy to reason about to have stateless-functional components

on the other hand, there are funny quirks about them (no backing store) and people are less familiar with them. I have stopped actively pushing for them even though I write them by default (and find that a good rule-of-thumb) but in this case the PR is changing it _away_ from what it was, a function, and to a component, but the reason it did that is now gone

Reply:
I tend to stick to `PureComponent` for some reason. No idea why, even tho `connect` makes them pure by default. Yeah, agree, we can go back to functional.

Reply:
changing it up to be a function component!

Reply:
fun fact! `PureComponent` slows us down when used with `connect()`, though probably in a trivial and insignificant way. the reason is that `connect()` already shallow-compares props, which is what `PureComponent` does by wrapping the component and adding the `shouldComponentUpdate()`

we can turn this off by passing options to `connect()` with `{ pure: false }`